### DISCUSSION TITLE
field state regardless of dirtiness

### DISCUSSION THREAD

**Discussion by DPflasterer** [react-hook-form/react-hook-form#9099] [General]
getFieldState.invalid returns false if a field has not yet been dirtied. In most cases this is exactly what I want. But in some cases I just want to know if an input with a default value is valid or not.

Is there any way to determine the fieldState validity regardless of it being touched/dirtied yet?

**Comment by bluebill1049** [comment]
`touchedFields` or `dirtyFields`?

**Reply by DPflasterer** [reply]
Is there a way to check if a field is valid at load? As in, before any input has been touched.

**Reply by bluebill1049** [reply]
not a field at the moment, but isValid is there to check.

**Reply by DPflasterer** [reply]
isValid does not accurately reflect the valid state of the field. A field is considered valid even when empty, so long as it has not been modified yet.

Since I'm using yup, I've been able to determine this by comparing getValues against the schema. It just seems like an unnecessary performance hit for something that's already being calculated.
